docs/src/tutorials/generating_reactions_programmatically.md

```diff
@@ -0,0 +1,152 @@
+## Population balance equations of the Smoluchowski coagulation model
+This tutorial shows how to programmatically construct a `ReactionSystem` corresponding to the chemistry underlying the [Smoluchowski coagulation model](https://en.wikipedia.org/wiki/Smoluchowski_coagulation_equation) using [ModelingToolkit](https://mtk.sciml.ai/stable/)/[Catalyst](https://catalyst.sciml.ai/dev/). A jump process version of the model is then constructed from the `ReactionSystem`, and compared to the model's analytical solution obtained by the [method of Scott](https://journals.ametsoc.org/view/journals/atsc/25/1/1520-0469_1968_025_0054_asocdc_2_0_co_2.xml) (see also mentioned in reference [3](https://doi.org/10.1006/jcph.2002.7017).
+
+The Smoluchowski coagulation equation describes a system of reactions in which single particles (singlets) may collide to form doublets, singlets and doublets may collide to form triplets, and so on. This models a variety of chemical/physical processes, including polymerization and flocculation.
+
+We begin by importing some necessary packages.
+```julia
+using ModelingToolkit, LinearAlgebra
+using DiffEqBase, DiffEqJump, OrdinaryDiffEq
+using LoopVectorization, Plots
+using BenchmarkTools
+using SpecialFunctions
+```
+Suppose the maximum cluster size is `N`. Lets initialize the system with some initial concentration `C`, initial number of singlets `uₒ` in the system. Since its a bimolecular chain of Reaction system(`nr` number of reactions), the bulk volume `V` of the system in which these binary collisions occur is important in the calculation of rate laws.
+  
+```julia
+## Parameter
+N = 10                       # maximum clusters size
+Vₒ = (4π/3)*(10e-06*100)^3   # volume of a singlet/monomer in cm³
```

Comment:
Are singlets and monomers the same? If so, I suggest using monomer, dimer, trimer, etc throughout and not using signlet/doublet/triplet, etc.

Reply:
> Are singlets and monomers the same?
>
Yes they are same..
 >If so, I suggest using monomer, dimer, trimer, etc throughout and not using signlet/doublet/triplet, etc.
>
I will make these changes